ait/gui/__init__.py

```diff
@@ -255,7 +255,14 @@ def handle(pathname):
         def handle(pathname):
             return bottle.static_file(pathname, root=HTMLRoot.User)
 
-        if host is None:
+        if hasattr(self, 'port'):
```

Comment:
I realize that it's not a problem in this instance but we should avoid `hasattr` in 2.x in my opinion. Of course, we're moving to 3.x shortly and it won't matter at all but sticking with `try ... except` here in 2.x is safer in the general case (it's also the same number of lines in this example). I also realize this is already present in the code so I don't blame you much =D. Don't need to change it, just a general comment for future consideration.

Soap boxing aside, isn't `getattr` cleaner here or is there a reason for the expanded check?

```
port = int(getattr(self, 'port', 8080))
```